### DISCUSSION TITLE
Is it possible to include multiple payments in a single cart/order?

### DISCUSSION THREAD

**Discussion by odhekar** [medusajs/medusa#5785] [Q&A]
Our scenario requires that the customer pays half of the cart total when placing order and the remaining half as cash on delivery(COD). We are using razorpay (custom PaymentProcessor similar to [this community project](https://github.com/SGFGOV/medusa-payment-razorpay)) for first payment and want cash on delivery for the second. I know we can use the medusa-payment-manual for the COD payment.

The payment intents for both providers are created in the cart as expected but then we have to make a choice and process the whole payment using a single payment method. 

Is this possible with built-in medusa features or should we be looking at forking medusa?

**Comment by lexvz14** [comment]
Hi there! 

Did you figure out a solution for this in the end?

**Reply by odhekar** [reply]
we ended up overriding [createPayment](https://github.com/medusajs/medusa/blob/develop/packages/medusa/src/services/payment-provider.ts#L471) - not a pretty solution but it works :)

**Reply by DexA1948** [reply]
Can you tell me how?
